test/unit/modules/compiler/sfc-parser.spec.js

```diff
@@ -30,6 +30,16 @@ describe('Single File Component parser', () => {
     expect(res.script.content.trim()).toBe('export default {}')
   })
 
+  fit('should parse template with closed input', () => {
```

Comment:
Yeah, I was just running this specific test while figuring out what is wrong. I know implementation for a fix fits somewhere in `compiler/parser/html-parser.js` but I'm not sure how to solve it.